DaS_Programs/Singly linked list/Singly linked list/main.c

```diff
@@ -328,15 +328,21 @@ struct node *delete_after(struct node *start)
         preptr=start;
         ptr=start->next;
         
-        while (preptr->data!=val)
+        while (ptr->next!=NULL&&preptr->data!=val)
         {
             preptr=ptr;
             ptr=ptr->next;
         }
-        preptr->next=ptr->next;
-        
-        free(ptr);
-    }
+        if(ptr->next==NULL)
```

Comment:
Okay this works. I understand you've put this here for a test case when the value to be deleted does not exist.

Reply:
yes